grasp_point[] grasp_points
